templates/api/deployment.yaml

```
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "farmdoc.fullname" . }}-api
+  labels:
+    {{- include "farmdoc.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.api.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "farmdoc.name" . }}-api
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "farmdoc.name" . }}-api
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- with .Values.api.image.pullSecrets | default .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: api
+          image: "{{ .Values.image.project }}/{{ .Values.api.image.repository }}:{{ .Values.api.image.tag }}"
+          imagePullPolicy: {{ .Values.api.image.pullPolicy | default .Values.image.pullPolicy }}
+          env:
+            - name: POSTGRES_USER
+              value: {{ .Values.api.env.postgres_user }}
+            - name: POSTGRES_PASSWORD
+              value: {{ .Values.api.env.postgres_password }}
```

we should get this from the secret:
```suggestion
              valueFrom:
                secretKeyRef:
                  name: my-secret
                  key: password
```

---

Changed to use secret